drivers/docker/driver.go

```diff
@@ -603,9 +603,9 @@ func (d *Driver) containerBinds(task *drivers.TaskConfig, driverConfig *TaskConf
 	secretDirBind := fmt.Sprintf("%s:%s", task.TaskDir().SecretsDir, task.Env[taskenv.SecretsDir])
 	binds := []string{allocDirBind, taskLocalBind, secretDirBind}
 
-	localBindVolume := driverConfig.VolumeDriver == "" || driverConfig.VolumeDriver == "local"
+	taskLocalBindVolume := driverConfig.VolumeDriver == ""
 
-	if !d.config.Volumes.Enabled && !localBindVolume {
+	if !d.config.Volumes.Enabled && !taskLocalBindVolume {
```

Comment:
Actually this might have a regression.  Can you double check behavior for interpolation in 0.8.7?  Looking at 0.8.7 code, if a user set volume driver to an absolute path to alloc dir (maybe through env-var interpolation?!), we would allow it.  I assume that's what the comment below is about:

> 		// Paths inside task dir are always allowed, Relative paths are always allowed as they mount within a container

Reply:
Never mind - `local` isn't allowed in - https://github.com/hashicorp/nomad/blob/v0.8.7/client/driver/docker.go#L1166 .  Maybe need to update comment then.  Paths inside task dir are allowed only if volume driver is empty.